### ISSUE TITLE
/__boot__.py:80: DeprecationWarning

### ISSUE THREAD

**Issue by kayihui** [ronaldoussoren/py2app#323]
Any idea how to resolve this?

Error msg:

myapp.app/Contents/Resources/__boot__.py:80: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
  import imp
Traceback (most recent call last):
  File "/dist/myapp.app/Contents/Resources/__boot__.py", line 489, in <module>
    _run()
  File "/dist/video_on_tkinter_v2.app/Contents/Resources/__boot__.py", line 483, in _run
    exec(compile(source, script, "exec"), globals(), globals())
  File "myapp.py", line 3, in <module>
    import PIL.Image, PIL.ImageTk
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 664, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 627, in _load_backward_compatible
  File "/dist/myapp.app/Contents/Resources/__boot__.py", line 86, in load_module
    return imp.load_module(
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/imp.py", line 244, in load_module
    return load_package(name, filename)
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/imp.py", line 216, in load_package
    return _load(spec)
  File "<frozen importlib._bootstrap>", line 710, in _load
AttributeError: 'NoneType' object has no attribute 'name'

**Comment by ronaldoussoren** [comment]
Can you reproduce this in a project that you can share?

**Comment by kayihui** [comment]
Hi there,

Sorry for the late reply.

You can find the project here:
https://github.com/kayihui/Mouse_app

I got the error message above with the following commend line code:
./dist/video_on_tkinter_v2.app/Contents/MacOS/video_on_tkinter_v2

Thank you!

**Comment by ronaldoussoren** [comment]
I found the problem: you're including a number of items in the packages section that don't have to be there and some of those are modules instead of packages. The latter causes the error.

The setup.py file below works for me:

```
from setuptools import setup

APP = ['video_on_tkinter_v2.py']
DATA_FILES = []
OPTIONS = {'argv_emulation':True,
           'packages':['cv2','pandas']}

setup(
    app=APP,
    options={'py2app': OPTIONS},
    setup_requires=['py2app'],
)
```

In general you should only explicitly add packages or modules to the application bundle when the automatic scan doesn't work (properly).  Py2app tries to do the right thing automatically by scanning your code (recursively) for import statements.  For some packages that doesn't work correctly, mostly when there are hidden imports (imports in C extensions, using the ``__import__`` function). In those cases the override can be used, and I often add special recipes for these to py2app.

**Comment by kayihui** [comment]
Hi there,

I tried with the setup.py you provided. But I got another error:

INTEL MKL ERROR: dlopen(/Users/kayihui/opt/anaconda3/lib/libmkl_intel_thread.dylib, 9): Library not loaded: @rpath/libiomp5.dylib
  Referenced from: /Users/kayihui/opt/anaconda3/lib/libmkl_intel_thread.dylib
  Reason: image not found.
Intel MKL FATAL ERROR: Cannot load libmkl_intel_thread.dylib.

It's my computer's problem???

Thank you so much for your time and effort!

**Comment by ronaldoussoren** [comment]
Ah, that's a bug I recently fixed.   There will be a new release later this week.

For now you can either install from this repository, or add "PIL" to the packages list.

**Comment by kayihui** [comment]
Hi there,

adding 'PIL' to the packages list doesn't seem to help. I got the same error message.

Anyway, thank you. 🙏

**Comment by ronaldoussoren** [comment]
The just released py2app 0.23 fixes this issue
